pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -406,7 +406,7 @@ func getOpenShiftNetworkConfig(ctx context.Context, c k8sclient.Client) (*osv1.N
 // Check whether running on OpenShift 4 by looking for operator objects that has been introduced
 // only in OpenShift 4
 func isRunningOnOpenShift4(c kubernetes.Interface) (bool, error) {
-	return isResourceAvailable(c, "configs", "imageregistry.operator.openshift.io", "v1")
+	return isResourceAvailable(c, "networks", "operator.openshift.io", "v1")
```

Comment:
If you look at node.status.images you have the openshift containers and there the openshift versions is present, maybe it's better to check that:

```yaml

      - docker.io/openshift/origin-node@sha256:7d5efa443cbfff23fb7ac965d50366b237ca4b0e8aabba57f765b58dfa1ddc58
      - docker.io/openshift/origin-node:v3.11.0
      sizeBytes: 1173010592
    - names:
 
```

Reply:
I think the right way to do it should be make a request to the api server with the `/version` and parse it just like we have the `virtctl version` and the `oc version` commands

Reply:
I agree this is not the best way to check it, but this PR is merely an RBAC bug https://github.com/kubevirt/cluster-network-addons-operator/issues/189 fix that has to be addressed quickly. It doesn't add any functionality, just extends ClusterRole (and on this commented line it switches the object so we can have less RBAC rules). Opened a new issue to track that https://github.com/kubevirt/cluster-network-addons-operator/issues/193.

Would you mind if we leave this as it is?

Reply:
Ack we can do a follow up to do proper versioning @SchSeba one is the way to go .